src/daemon/ca/manager.rs

```diff
@@ -887,92 +904,236 @@ impl CaManager {
+                                        handle,
+                                        rcn.clone(),
+                                        RcvdCert::from(issued),
+                                        self.config.clone(),
+                                        self.signer.clone(),
+                                        actor,
+                                    ))
+                                    .await {
+                                        // Note that sending the command to update a received certificate
+                                        // cannot fail unless there are bigger issues like this being the wrong
+                                        // response for this resource class. This would be extremely odd because
+                                        // we only just asked the resource class which request to send. Still, in
+                                        // order to handle this the most graceful way we can, we should just drop
+                                        // this resource class and report an error. If there are are still resource
+                                        // entitlements under the parent for this resource class, then a new class
+                                        // will be automatically created when we synchronize the entitlements again.
+
+                                        let reason = format!("received certificate cannot be added, error: {}", e);
+
+                                        self.send_command(CmdDet::drop_resource_class(
+                                            handle, 
+                                            rcn.clone(), 
+                                            reason.clone(), 
+                                            self.signer.clone(),
+                                            actor,
+                                        )).await?;
+
+                                        // push the error for reporting, this will also trigger that the CA will
+                                        // sync with its parent again - and then it will just find revocation
+                                        // requests for this RC - which are sent on a best effort basis
+                                        errors.push(Error::CaParentSyncError(
+                                            handle.clone(),
+                                            parent.clone(),
+                                            rcn.clone(),
+                                            reason,
+                                        ));
+                                        break;
+                                    }
+                                },
+                                rfc6492::Res::NotPerformed(not_performed) => {
+                                    match not_performed.status() {
+                                        1201 | 1202 => {
```

Comment:
Note what 1201 and 1202 mean here, like was done above for 1301 and 1302?

Reply:
These error codes are defined in RFC 6492. The comments explain what the server told us in more understandable language. An enum could have been used when defining ``NotPerformed`` but we didn't do this at the time. I do not see a strong case for doing it now though because these values will not change.

Reply:
Okay, but anyone reading this is going to have to know what 1201 means and for that they'll have to look at the spec, isn't it easier for future maintainers just to mention what 1201 means like you did above for 1301 and 1302? Or am I missing something?

Reply:
Sure, the text below already explained the meaning but I have added an explicit mention of each error code. (Note and please ignore that the code was also reformatted as a result of saving it in my IDE)